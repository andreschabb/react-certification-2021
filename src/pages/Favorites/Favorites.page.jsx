import React, { useState, useEffect } from 'react';
import _ from 'underscore';

import VideoList from '../../components/VideoList';
import { FAV_STORAGE_KEY } from '../../utils/constants';
import { storage } from '../../utils/storage';
import { Container, Title } from './Favorites.styles';

const FavoritesPage = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    const myFavs = storage.get(FAV_STORAGE_KEY);
    if (myFavs) {
      setVideos(_.flatten(_.values(myFavs)));
    } else {
      setVideos(null);
    }
  }, []);

  return videos && videos.length > 0 ? (
    <Container fixed={videos.length < 5}>
      <Title>My Favorites</Title>
      <VideoList videos={videos} />
    </Container>
  ) : (
    <Container fixed>
      <Title>There are no videos on My Favorites</Title>
    </Container>
  );
};

export default FavoritesPage;
