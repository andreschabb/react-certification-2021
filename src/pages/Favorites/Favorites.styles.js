import styled from 'styled-components';

export const Container = styled.section`
  height: ${({ fixed }) => (fixed ? '85vh' : '100%')};
  width: 80%;
  margin-bottom: 5%;
  text-align: center;
`;

export const Title = styled.h1`
  font-size: 4rem;
  letter-spacing: -6px;
  font-family: 'Lucida Console', 'Courier New', monospace;
  font-weight: normal;

  @media screen and (max-width: 750px) {
    font-size: 2rem;
    letter-spacing: -4px;
  }

  @media screen and (max-width: 500px) {
    font-size: 1.5rem;
    letter-spacing: -2px;
  }
`;
