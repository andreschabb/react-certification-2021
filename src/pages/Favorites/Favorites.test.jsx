import React from 'react';
import { render, screen } from '@testing-library/react';

import Favorites from './Favorites.page';

describe('Favorites', () => {
  it('should renders', () => {
    const { container } = render(<Favorites />);

    expect(screen.getByText('My Favorites')).toBeTruthy();

    expect(container).toMatchSnapshot();
  });
});
