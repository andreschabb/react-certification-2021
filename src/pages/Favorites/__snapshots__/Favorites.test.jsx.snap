// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should renders 1`] = `
<div>
  <section
    class="sc-iBPRYJ tFyuG"
  >
    <h1
      class="sc-fubCfw hnRPzm"
    >
      My Favorites
    </h1>
    <div
      class="sc-gKsewC ikKpsV"
      data-testid="video-list"
    />
  </section>
</div>
`;
